File: analysis/pricedata.py
from dataclasses import dataclass
from datetime import datetime


@dataclass
class PriceData:
    current_off_peak: float = 0.067
    current_peak: float = 28.26
    current_off_peak_start: str = "00:00"
    current_off_peak_stop: str = "00:07"
    current_start: datetime = datetime.strptime("2025-07-10", "%Y-%m-%d")
    current_stop: datetime = datetime.strptime("2026-09-10", "%Y-%m-%d")
    off_peak_average: float = 0.96         # computed by EnergySummary; do not set directly
    off_peak_baseline_kwh: float = 0.96   # expected kWh/day at off-peak for a 7-hour window
    current_export: float = 0.165
    standing_charge: float = 0.6
    original_price: float = 6210 
    interest_rate: float = 3.7

File: android/app/src/main/java/com/richw/sunsynk/analysis/PriceData.kt
package com.richw.sunsynk.analysis

import java.time.LocalDate

data class PriceData(
    var currentOffPeak: Double = 0.067,
    var currentPeak: Double = 28.26,
    var currentOffPeakStart: String = "00:00",
    var currentOffPeakStop: String = "00:07",
    var currentStart: LocalDate = LocalDate.of(2025, 7, 10),
    var currentStop: LocalDate = LocalDate.of(2026, 9, 10),
    var offPeakAverage: Double = 0.96,
    var offPeakBaselineKwh: Double = 0.96,
    var currentExport: Double = 0.165,
    var standingCharge: Double = 0.6,
    var originalPrice: Double = 6206.47,
    var interestRate: Double = 3.7,
)
